worker/src/main/java/gov/cms/ab2d/worker/service/FileServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package gov.cms.ab2d.worker.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static java.nio.file.StandardOpenOption.APPEND;
+
+@Slf4j
+@Component
+public class FileServiceImpl implements FileService {
+
+
+    @Override
+    public Path createDirectory(Path outputDir) {
+        Path outputDirectory = null;
+        try {
+            outputDirectory = Files.createDirectories(outputDir);
+        } catch (IOException e) {
+            final String errMsg = "Could not create output directory : ";
+            log.error("{} : {}", errMsg, outputDir.toAbsolutePath());
+            throw new UncheckedIOException(errMsg + outputDir.getFileName(), e);
+        }
+        return outputDirectory;
+    }
+
+
+    @Override
+    public Path createFile(final Path outputDir, final String filename) {
```

Comment:
Since I didn't realize you'd delete the file if it existed from the name, maybe call it overwriteFile? And then give an error for not being able to replace file or create file.

Reply:
So, my understanding of the scenario is this. This should ordinarily not happen.

At the beginning of the batch process, it will create a directory (using the jobUuid) and an output file inside it.

If the batch were to fail due to some reason and is re-run, the file would exist from the _failed run_. 
To cover this scenario, Denis suggested that I put the `Files.deleteIfExists`  line. It currently does throw an `UncheckedIOException`  if it could not create the output file

Reply:
I have no problem with the logic, just the name. The default behavior of "create" is to fail if the file already exists. This is more of a createOrReplace. If anyone else wants to use this method, the method name is misleading.

Reply:
I see what you mean. I've renamed the method to `createOrReplaceFile`

Reply:
Thanks.